include/ginkgo/core/base/math.hpp

```diff
@@ -1220,10 +1220,14 @@ GKO_INLINE GKO_ATTRIBUTES T safe_divide(T a, T b)
  * @return `true` if the value is NaN.
  */
 template <typename T>
-GKO_INLINE GKO_ATTRIBUTES std::enable_if_t<!is_complex_s<T>::value, bool>
-is_nan(const T& value)
+GKO_DEPRECATED(
+    "is_nan can't be used safely on the device (MSVC+CUDA), and will thus be "
+    "removed in a future release, without replacement")
```

Comment:
We only use this function in trs apparently, so I am fine to deprecate this now.
If we have other usage, which does not rely on our initial value, I will prefer still providing a proper one (std::is_nan in general but a bit time/memory-consuming on msvc+cuda side)